libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,44 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	firewalldRunning, err = checkRunning()
+	if err != nil {
+		return err
+	}
+	// If firewalld is not running on the first check poll
+	if !firewalldRunning {
+		tick := time.NewTicker(500 * time.Millisecond)
+		startTime := time.Now()
+		defer tick.Stop()
+
+	TickLoop:
+		for {
+			select {
+			case <-tick.C:
+				firewalldRunning, err = checkRunning()
+				if err != nil {
+					log.G(context.TODO()).Debugf("Check Running failed: %v", err)
+				}
+
+				log.G(context.TODO()).Debugf("Checking status of Firewalld")
+
+				if firewalldRunning || time.Since(startTime) >= fireWalldInitializationTimeout {
+					break TickLoop
+				}
+
+			default:
+				continue TickLoop
```

Comment:
Why have the `default` case? A `select` with a `default` case never blocks. Consequently, this is effectively a tight polling loop for the ticker to tick.